subxt/src/utils/unchecked_extrinsic.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! The "default" Substrate/Polkadot UncheckedExtrinsic.
+//! This is used in codegen for runtime API calls.
+//!
+//! The inner bytes represent the encoded extrinsic expected by the
+//! runtime APIs. Deriving `EncodeAsType` would lead to the inner
+//! bytes to be re-encoded (length prefixed).
+
+use std::marker::PhantomData;
+
+use codec::Decode;
+use scale_decode::{visitor::DecodeAsTypeResult, IntoVisitor, Visitor};
+
+/// The unchecked extrinsic from substrate.
+#[derive(Clone, Debug, Eq, PartialEq, Decode)]
+pub struct UncheckedExtrinsic<Address, Call, Signature, Extra>(
+    pub Vec<u8>,
+    #[codec(skip)] pub PhantomData<(Address, Call, Signature, Extra)>,
+);
+
+impl<Address, Call, Signature, Extra> codec::Encode
+    for UncheckedExtrinsic<Address, Call, Signature, Extra>
+{
+    fn encode(&self) -> Vec<u8> {
+        self.0.to_owned()
+    }
+}
+
+impl<Address, Call, Signature, Extra> scale_encode::EncodeAsType
+    for UncheckedExtrinsic<Address, Call, Signature, Extra>
+{
+    fn encode_as_type_to(
+        &self,
+        _type_id: u32,
+        _types: &scale_info::PortableRegistry,
+        out: &mut Vec<u8>,
+    ) -> Result<(), scale_encode::Error> {
+        out.extend_from_slice(&self.0);
+        Ok(())
+    }
+}
+
+impl<Address, Call, Signature, Extra> scale_encode::EncodeAsFields
+    for UncheckedExtrinsic<Address, Call, Signature, Extra>
+{
+    fn encode_as_fields_to(
+        &self,
+        _fields: &mut dyn scale_encode::FieldIter<'_>,
+        _types: &scale_info::PortableRegistry,
+        out: &mut Vec<u8>,
+    ) -> Result<(), scale_encode::Error> {
+        out.extend_from_slice(&self.0);
+        Ok(())
+    }
+}
```

Comment:
I'm not sure it makes sense to have this impl, since we can't sensible encode the thing to any set of fields (and the target type is an "UncheckedExtrinsic" type anyway and not some fields.

Does everything work ok without it? If so let's remove it :)